contracts/truefi2/CollateralVault.sol

```diff
@@ -43,13 +51,13 @@ contract CollateralVault is ICollateralVault, UpgradeableClaimable {
     }
 
     function stake(uint256 amount) external {
-        // require(borrowingMutex.isUnlocked(borrower) || borrowingMutex.locker(borrower) == LOCA)
-        // if borrowingMutex.locker(borrower) == LOCA:
-        //   poke LOCA for borrower to update limit + rate
-        // safe transfer amount from borrower
-        require(amount == 0); // silence lint
-        stakedToken = IERC20WithDecimals(address(0)); // silence build warning
-        revert("Unimplemented!");
+        require(
+            borrowingMutex.isUnlocked(msg.sender) || borrowingMutex.locker(msg.sender) == address(lineOfCreditAgency),
+            "CollateralVault: Borrower cannot stake when they have an ongoing fixed term loan"
```

Comment:
Let's keep error messages in sync with local code. We'll introduce new agencies in the future but won't remember to update this error message.
```suggestion
            "CollateralVault: Borrower can only stake when they're unlocked or have a line of credit"
```